### ISSUE TITLE
Race condition in time reset for sensors

### ISSUE THREAD

**Issue by osrf-migration** [osrf/gazebo#236]
**Original report ([archived issue](https://osrf-migration.github.io/gazebo-gh-pages/#!/osrf/gazebo/issues/236)) by Steve Peters (Bitbucket: [Steven Peters](https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/), GitHub: [scpeters](https://github.com/scpeters)).**

----------------------------------------
Sensors may not be successfully reset by a world_control message. There is a race condition between the Sensor::OnControl and World::OnControl functions since they are in different threads. If Sensor::OnControl executes first, then each sensor's lastUpdateTime will be reset to 0. However, a Sensor::Update may run before World::OnControl executes, which may increase the value of lastUpdateTime to the current sim time.

Basically, the sensors shouldn't be reset until the World has been reset, and these OnControl callbacks don't guarantee that.
